File: javascript/book_problems/01_greedy/02_곱하기혹은더하기.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input;
rl.on("line", (line) => {
  input = line.split("").map((value) => parseInt(value));
  rl.close();
}).on("close", () => {
  const result = getBiggest(input);
  console.log(result);
  process.exit(0);
});

function getBiggest(input) {
  for (let i = 0; i < input.length - 1; i++) {
    const a = input[i];
    const b = input[i + 1];

    input[i + 1] = Math.max(a + b, a * b);
  }

  return input[input.length - 1];
}

File: python/book_problems/02_곱하기혹은더하기.py
arr = list(map(int, list(input())))

print(arr)

result = arr[0]

for i in range(1, len(arr)):
    sum_value = arr[i]+result
    mul_value = arr[i]*result
    print(arr[i], result, sum_value, mul_value)

    result = max(sum_value, mul_value)

print(result)
